### PR TITLE
Project Summary | Render project extent map

### PR DESCRIPTION
**Pull request by mddilley**
Closes https://github.com/cityofaustin/atd-data-tech/issues/4749
Closes https://github.com/cityofaustin/atd-data-tech/issues/4675

This PR adds a map to visualize features associated with a project in the Project Summary view. I updated the way that layers are selected and presented by creating a configuration per layer source. This unifies the presentation of the layers in New Project workflow and the Project Summary view based on the source and its associated configuration (color, etc.). I also added some docstrings to the functions in `mapHelpers.js`.

For some draft documentation on Moped maps, there is a new User Interface section in the Infrastructure group in the Gitbook Moped Technical docs with a "Maps" page.

### Selecting in New Project stepper
Screen Shot 2020-12-28 at 1 52 07 PM

### Project Summary view after saving
Screen Shot 2020-12-29 at 11 41 37 AM

### CODE REVIEW

File: moped-editor/src/utils/mapHelpers.js [hunk 1]
```diff
@@ -2,23 +2,53 @@ import React from "react";
 import theme from "../theme/index";
 import { isEqual } from "lodash";
 
-export const MAPBOX_TOKEN = `pk.eyJ1Ijoiam9obmNsYXJ5IiwiYSI6ImNraWV4dHR0ZjAwNnYyd3FwYjFoNHduaDcifQ.--3vRm2KHq1gh5K_L0pqtA`;
+export const MAPBOX_TOKEN = process.env.REACT_APP_MAPBOX_TOKEN;
```
Comment (mateoclarke) [inline moped-editor/src/utils/mapHelpers.js:5]: 👍

File: moped-editor/src/utils/mapHelpers.js [hunk 2]
```diff
@@ -32,57 +62,114 @@ export const getGeoJSON = e =>
...
+ * @return {Boolean} Is feature present in features of feature collection in state
+ */
 export const isFeaturePresent = (selectedFeature, features) =>
   features.some(feature => isEqual(selectedFeature, feature));
 
-export const createProjectLayerConfig = (
-  polygonId,
-  layerSourceName,
+/**
+ * Create a configuration to set the Mapbox spec styles for selected/unselected/hovered layer features
```
Comment (mateoclarke) [inline moped-editor/src/utils/mapHelpers.js:75]: nice!!! love all these docstrings! Ignore what I said in my last PR review. You're way ahead of me on this!

File: moped-editor/src/views/projects/projectView/ProjectSummary.js [hunk 3]
```diff
@@ -94,14 +96,24 @@ const ProjectSummary = () => {
...
+                <Box mb={2}>
+                  <h4>{detail.label}</h4>
+                  <p>{formatValue(detail)}</p>
+                </Box>
+              </Grid>
+            ))}
           </Grid>
-        ))}
+        </Grid>
+        <Grid item xs={6}>
```
Comment (mateoclarke) [inline moped-editor/src/views/projects/projectView/ProjectSummary.js:111]: I think we need to maybe update these side-by-side components to stack on smaller screen widths. maybe xs=12 and sm or md=6?

File: moped-editor/src/utils/mapHelpers.js [hunk 4]
```diff
@@ -2,23 +2,53 @@ import React from "react";
 import theme from "../theme/index";
 import { isEqual } from "lodash";
 
-export const MAPBOX_TOKEN = `pk.eyJ1Ijoiam9obmNsYXJ5IiwiYSI6ImNraWV4dHR0ZjAwNnYyd3FwYjFoNHduaDcifQ.--3vRm2KHq1gh5K_L0pqtA`;
+export const MAPBOX_TOKEN = process.env.REACT_APP_MAPBOX_TOKEN;
 
 export const mapInit = {
   latitude: 30.268039,
   longitude: -97.742828,
   zoom: 12,
```
Comment (mateoclarke) [inline moped-editor/src/utils/mapHelpers.js:10]: I think of the summary page, the map should be more zoomed in by default. 12 seems like a good zoom level for the project extent step in create new project. But when we're talking about a specific existing project, more detail is probably better.

**Review by mateoclarke — APPROVED** [review]
🚢 🗺️🚢 🗺️🚢 🗺️ 🚢 

This looks so good! Can't wait for the next demo!

anne_hathaway_clapping

**Comment by mateoclarke** [comment]
Didn't catch these docs until I re-read the description, love it!
https://app.gitbook.com/@atd-dts/s/moped-technical-docs/infrastructure/user-interface/maps

**Review by sergiogcx — CHANGES_REQUESTED** [review]
It looks like it breaks whenever you open a project that does not have any geometry data:





```
×
TypeError: Cannot convert undefined or null to object
sumFeaturesSelected
src/utils/mapHelpers.js:173
  170 |  * @param {Object} selectedLayerIds - An object whose keys are layer names and values are arrays of ID strings
  171 |  * @return {Number} Total number of string IDs
  172 |  */
> 173 | export const sumFeaturesSelected = selectedLayerIds =>
  174 |   Object.values(selectedLayerIds).reduce(
  175 |     (acc, selectedIds) => (acc += selectedIds.length),
  176 |     0
View compiled
ProjectSummaryDetailsMap
src/views/projects/projectView/ProjectSummaryDetailsMap.js:82
  79 |     )}
  80 |     {renderTooltip(hoveredFeature, hoveredCoords, classes.toolTip)}
  81 |   </ReactMapGL>
> 82 |   <Typography className={classes.locationCountText}>
     | ^  83 |     {sumFeaturesSelected(selectedLayerIds)} locations in this project
  84 |   </Typography>
  85 | </Box>
View compiled
```

Otherwise, it works when a project is created. It looks amazing 🙌 🎖️ 💯

**Review by amenity — APPROVED** [review]
I can't turn up the Netlify link, but looks these screenshots (& docs!) look 💯!

File: moped-editor/src/views/projects/projectView/ProjectSummary.js (hunk 3 above)
Comment (mddilley) [reply moped-editor/src/views/projects/projectView/ProjectSummary.js:111]: This is a really good point, and I'll update before merge so we keep these fields and map looking sharp on smaller screens!

File: moped-editor/src/utils/mapHelpers.js (hunk 4 above)
Comment (mddilley) [reply moped-editor/src/utils/mapHelpers.js:10]: This is a great idea! 💡 I created an issue to auto-zoom to the features on the map. We did something similar in the Signs & Markings React map so it shouldn't be too complex. 🤞 https://github.com/cityofaustin/atd-data-tech/issues/4819

**Pull request merged**